### PR TITLE
Migrate continuous integration from Travis CI to GitHub Actions

### LINKED ISSUE #30: Migrate continuous integration to GitHub Actions
We'll lose the use of the Travis workers at some point in the near future (current target is the end of August). It would be good to migrate the continuous integration to GitHub Actions before that point.

### PR DESCRIPTION
**Pull request by mdickinson**
This PR adds two GitHub Actions workflows

- A per-commit workflow that runs the tests on each commit to an open PR
- A weekly cron job that runs the test suite against the latest ETS sources

We also remove the existing Travis CI configuration.

The new workflows mostly replicate the existing Travis CI config, except that we're not uploading coverage. Let me know if the coverage upload is useful.

Closes #30

### CODE REVIEW

File: .github/workflows/validate-pr.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+on:
+  pull_request
+
+jobs:
+  test-with-edm:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        toolkit: ['pyqt', 'pyqt5', 'pyside2']
```
Comment (rahulporuri) [inline .github/workflows/validate-pr.yml:20]: We've stopped testing against pyqt in other ETS projects and we can probably stop here too. I'll defer the final decision to @stevenjkern .

File: .github/workflows/weekly-check.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    # Run at 01:48 UTC every Friday
+    - cron: '48 1 * * 5'
+
+jobs:
+  test-with-edm:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        toolkit: ['pyqt', 'pyqt5', 'pyside2']
```
Comment (rahulporuri) [inline .github/workflows/weekly-check.yml:23]: same comment here regd not testing with `pyqt`

**Review by rahulporuri — APPROVED** [review]
Almost LGTM.

The `install-edm-*` scripts can also be removed - https://github.com/enthought/graphcanvas/blob/837f601713e8162365b8c05d4d05ee410b50f78d/ci/install-edm-linux.sh and https://github.com/enthought/graphcanvas/blob/837f601713e8162365b8c05d4d05ee410b50f78d/ci/install-edm-osx.sh

I looked at the per-PR CI logs (on this PR) and I also looked at the weekly cron job logs and they all look good/as expected.

**Comment by mdickinson** [comment]
Updated to remove the EDM installation scripts. Leaving `pyqt` was a deliberate decision - I didn't want to introduce lots of changes between the old Travis CI and the new GitHub Actions CI during the migration; I'd rather make those changes once we've migrated. (Though the removal of coverage is a counterexample, of course.)

**Review by stevenjkern — APPROVED** [review]
LGTM :shipit:

**Pull request merged**